pallets/pallet-did-lookup/src/migrations.rs

```diff
@@ -16,74 +16,144 @@
+		} else {
+			log::info!("🔎 DidLookup: Initiating migration");
+			let mut connected_dids = 0u64;
+			let mut connected_accounts = 0u64;
+
+			// Migrate connected DIDs
+			// We should not write to the same storage item during drain because it can lead
+			// to undefined results. Thus, we write to a temporary storage and move that at
+			// the end. Else we iterate over every key more or less twice.
+			ConnectedDids::<T>::drain().for_each(|(acc_id32, value)| {
+				log::debug!("🔎 #{} Migrating ConnectedDid for account", connected_dids,);
+				let acc_id: LinkableAccountId = acc_id32.into();
+				TmpConnectedDids::<T>::insert(acc_id, value);
+				connected_dids = connected_dids.saturating_add(1);
+			});
+			log::info!("🔎 DidLookup: Migrated all {} ConnectedDids", connected_dids);
+
+			// Migrate accounts
+			ConnectedAccounts::<T>::drain().for_each(|(did_id, acc_id32, val)| {
+				log::debug!("🔎 #{:?} Migrating ConnectedAccount", connected_accounts,);
+				let acc_id: LinkableAccountId = acc_id32.into();
+				TmpConnectedAccounts::<T>::insert(did_id, acc_id, val);
+				connected_accounts = connected_accounts.saturating_add(1);
 			});
+			log::info!("🔎 DidLookup: Migrated all {:?} ConnectedAccounts", connected_accounts);
 
-		Pallet::<T>::current_storage_version().put::<Pallet<T>>();
+			// Move TmpStorage
+			move_storage::<Pallet<T>>(b"TmpConnectedDids", b"ConnectedDids");
+			move_storage::<Pallet<T>>(b"TmpConnectedAccounts", b"ConnectedAccounts");
 
+			Pallet::<T>::current_storage_version().put::<Pallet<T>>();
+
+			<T as frame_system::Config>::DbWeight::get().reads_writes(
+				(connected_dids.saturating_add(connected_accounts)).saturating_mul(2),
+				(connected_dids.saturating_add(connected_accounts))
+					.saturating_mul(2)
+					.saturating_add(1),
+			)
+		}
+	}
+
+	#[cfg(feature = "try-runtime")]
+	fn pre_upgrade() -> Result<(), &'static str> {
+		assert_eq!(Pallet::<T>::on_chain_storage_version() < Pallet::<T>::current_storage_version());
+
+		// Store number of connected DIDs in temp storage
+		let connected_did_count: u64 = ConnectedDids::<T>::iter_keys().count().saturated_into();
+		Self::set_temp_storage(connected_did_count, "pre_connected_did_count");
 		log::info!(
-			"👥  completed DID lookup pallet migration to {:?} ✅",
-			Pallet::<T>::current_storage_version()
+			"🔎 DidLookup pre migration: Number of connected DIDs {:?}",
+			connected_did_count
 		);
 
-		total_weight
+		// Store number of connected accounts in temp storage
+		let connected_account_count: u64 = ConnectedAccounts::<T>::iter_keys().count().saturated_into();
+		Self::set_temp_storage(connected_account_count, "pre_connected_account_count");
+		log::info!(
+			"🔎 DidLookup pre migration: Number of connected accounts {:?}",
+			connected_account_count
+		);
+		Ok(())
 	}
 
 	#[cfg(feature = "try-runtime")]
 	fn post_upgrade() -> Result<(), &'static str> {
+		assert_eq!(Pallet::<T>::on_chain_storage_version() == Pallet::<T>::current_storage_version());
```

Comment:
Same here.

Either

```suggestion
		assert_eq!(Pallet::<T>::on_chain_storage_version(), Pallet::<T>::current_storage_version());
```

or

```suggestion
		assert!(Pallet::<T>::on_chain_storage_version() == Pallet::<T>::current_storage_version());
```